### DISCUSSION TITLE
Search for all models/objects/time series

### DISCUSSION THREAD

**Discussion by kfollesdal** [Volue-Public/energy-mesh-python#355] [Q&A] [answered]
I have connected to the Mesh server with the API. I can get time series data when I now the key. But I can not find the same time series with search even if I now the path, only get and empty list back. 

I want to search/view all models/objects/time series on the server. Is the a root path where we can start a query from with a wildcard search? I have tried with path "" and "/" and query "*". But no result.

**Comment by tnoczyns-volue** [comment, marked as answer]
Hi,

I extended the `timeseries_search` example with different approaches and descriptions that could serve your needs. See PR: https://github.com/Volue-Public/energy-mesh-python/pull/357.

The wildcard "*" works on object level, where it returns all objects, but not on attribute level. See documentation for more:
https://volue-public.github.io/energy-mesh-python/mesh_search.html

Look at the above mentioned example to see how you could list all of time series attributes.

For the getting the root object/model I added an issue https://github.com/Volue-Public/energy-mesh-python/issues/356. 

In some cases the `EnergySystem` model name is just the name of the client company. You can verify by accessing the server where Mesh is running, go to the binaries of Mesh and run a tool `IntMeshQuery.exe` without any parameters. It will list all of the model definitions and their instances (i.e. models):

```console
C:\Mesh\bin>IntMeshQuery.exe
Mesh Client Serialization version 24 (max supported is 24)
connecting to Mesh server at tcp://127.0.0.1:40321 (timeout: 30)...
connected
Connected and ready: 53 ms

Action:RepositoryModels
EnergySystem
. ModelName
System
. Utility
```

In this case the `ModelName` is the name you're looking for. You can use it as the `start_object_path` in the `timeseries_search` example in `search_method_2` like: `Model/ModelName`.

Please let me know if something is unclear.

Regards,
Tomasz

**Reply by kfollesdal** [reply]
@tnoczyns-volue thank you for the explanation and examples.

**Comment by tnoczyns-volue** [comment]
Hi,

The functionality for getting root object/model (#356) was added to Python SDK and requires Mesh 2.10 that will be released beginning of September.

Regards,
Tomasz

**Reply by kfollesdal** [reply]
Very good. Looking forward to the new version.
